### ISSUE TITLE
Feature request: Add basic client side load balancing for gRPC

### ISSUE THREAD

**Issue by vermas7988** [apache/pekko-grpc#417]
There is no client side load balancing for gRPC module. 
All the requests land on one instance rather than distributing the load between multiple instances.
Is it possible to add basic load balancing on client side e.g. round_robin or other pattern so request load can be distributed among instances?

**Comment by pjfanning** [comment]
That akka-grpc PR is in Pekko gRPC. The fork was made after the merge was made.

**Comment by vermas7988** [comment]
indeed, but there is no clear doc/explaination on how to achieve client side load balancing, even with the default static mechanism as I mentioned in above comment.

**Comment by pjfanning** [comment]
I will repeat pekko-grpc is built on top of grpc-java and https://pekko.apache.org/docs/pekko-grpc/current/client/details.html#load-balancing highlights how to set the load-balancing-policy config. You can go and read the grpc-java docs to find out what the settings are. Apache Pekko is a free lib with no warranties or support. You can help by coming back with what you learn and add to the pekko-grpc docs to help future users.

**Comment by raboof** [comment]
> As mentioned in the link you shared, "_When multiple endpoints are discovered for a gRPC client, currently one is selected and used for all outgoing calls_".

We should probably add a 'by default' there.

> Ideally we should be able to send requests to both endpoints in a `round_robin` fashion. Even if `round_robin` is provided as load balancing policy, only one host is receiving requests.

As that documentation mentions, while perhaps counter-intuitive, there are situations where that's a better strategy - but I agree it would be good if client-side load balancing were possible.

> I see in the docs, "_Client-side load balancing is desirable when you are using the default static or the grpc-dns discovery mechanism._".

I suspect we should replace 'desirable' with 'possible' here.

> But the meaning of default static method is not clear to me.

'static' is the discovery method that is used when you don't configure a different service discovery mechanism (https://pekko.apache.org/docs/pekko-grpc/current/client/configuration.html#using-pekko-discovery-for-endpoint-discovery).

> Can we add some example for this use case?

I agree that would be helpful.

> Even if `round_robin` is provided as load balancing policy, only one host is receiving requests.

Indeed that seems to contradict the documentation. Could you boil down your application to a minimal test project to show the behavior?
